### ISSUE TITLE
Reproducing celebA results

### ISSUE THREAD

**Issue by rhngla** [Gabe-YHLee/IRVAE-public#1]
I'm trying to reproduce results reported in Table 1 (celebA P@k) in your [paper](https://openreview.net/pdf?id=mQxt8l7JL04), and referring to the description in `D.3 Section 5.2`.

1. Based on the description in `D.3 Section 5.2`, the final encoder layer is `Conv2d (1024, 256, 1, 1, 0)`, which seems off; what am I missing?
2. Within the ELBO loss, what is the scale factor for the Gaussian likelihood (reconstruction)?
3. How are the train/validation/test splits chosen?

Cool paper, and appreciate any help in this regard!

**Comment by rhngla** [comment]
Author responded by email and I was able to roughly replicate the experiment. 

1. The input is `3, 64, 64` dimensional. With the architecture described in the paper, it leads to a `256` dimensional output from the encoder layer, which needs to be split into `128` dimensional `mu` and `logvar`. Code to transform the raw celebA data: 
```python
import torchvision
import torchvision.transforms as transforms

transform = transforms.Compose([
    transforms.Resize(64),
    transforms.CenterCrop(64),
    transforms.ToTensor(),
    transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))])

celeba_data_train = torchvision.datasets.CelebA(data_path, 
                transform=transform, 
                split='train', download=False)
```
2. The scale is set to 1.0

```python
def gauss_lhood(x_hat, x, sd=1.0):
    dist = torch.distributions.Normal(loc=x_hat, scale=sd)
    log_pxz = dist.log_prob(x)
    return torch.sum(log_pxz, dim=(1, 2, 3))
```

3. The `split` option in `torchvision.datasets.CelebA` is used to specify the train/validation/test splits. 

Finally, my numbers for the plain VAE results reported in Table 1 of the manuscript:

| k   | P@k (SAR) |  P@k (DAR) |
| --- | -------- | -------- |
| 1   | 0.70  | 0.49 |
| 5   | 0.70  | 0.46 |
| 10  | 0.69  | 0.44 |
| 20  | 0.67  | 0.42 |
